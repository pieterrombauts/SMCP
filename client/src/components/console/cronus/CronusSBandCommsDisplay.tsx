import React from 'react';
import styled from 'styled-components';
import { cronusSBandDisplay } from '../customTypes'

interface CronusSCDProps {
  className?: string;
  values: cronusSBandDisplay;
};

const UCronusSBandCommsDisplay: React.FC<CronusSCDProps> = ( props ) => {
  return(
    <div className={props.className}>
      <svg preserveAspectRatio='xMidYMid meet' viewBox='0 0 1300 990'>
        <image x={0} y={0} width={1300} height={990} href='/media/Cronus_3.png'></image>
        <rect x="450" y="380" rx="5" ry="5" width="450" height="385" style={{fill:"#36485b",stroke:"#2a3a3f",strokeWidth:"5"}} />
                
        <text x="38%" y="45%" fill="white" font-size="45" font-family="Verdana">S-Band Status</text>

        <text x="38%" y="51%" fill="white" font-size="20" font-family="Verdana">Active S-Band</text>
        <text x="58%" y="51%" fill="#d6ab62" font-size="20" font-family="Verdana">1</text>

        <text x="38%" y="56%" fill="white" font-size="20" font-family="Verdana">S-Band1 Power</text>
        <text x="58%" y="56%" fill="#d6ab62" font-size="20" font-family="Verdana">On</text>

        <text x="41%" y="59%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
        <text x="58%" y="59%" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.sBand.elevationGimbal}</text>
        <text x="63%" y="59%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="41%" y="62%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
        <text x="58%" y="62%" fill="#d6ab62" font-size="20" font-family="Verdana">{props.values.sBand.azimuthGimbal}</text>
        <text x="63%" y="62%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="38%" y="67%" fill="white" font-size="20" font-family="Verdana">S-Band2 Power</text>
        <text x="58%" y="67%" fill="#d6ab62" font-size="20" font-family="Verdana">Off</text>

        <text x="41%" y="70%" fill="white" font-size="20" font-family="Verdana">Elevation Gimbal</text>
        <text x="58%" y="70%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
        <text x="63%" y="70%" fill="white" font-size="20" font-family="Verdana">째</text>

        <text x="41%" y="73%" fill="white" font-size="20" font-family="Verdana">Azimuth Gimbal</text>
        <text x="58%" y="73%" fill="#d6ab62" font-size="20" font-family="Verdana">0.00</text>
        <text x="63%" y="73%" fill="white" font-size="20" font-family="Verdana">째</text>
      </svg>
    </div>
  )
};

const CronusSBandCommsDisplay = styled(UCronusSBandCommsDisplay)`
  svg {
    width: 100%;
    height: 100%;
  }
`;

export default CronusSBandCommsDisplay;