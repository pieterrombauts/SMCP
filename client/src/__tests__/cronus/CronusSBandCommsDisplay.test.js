import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import CronusSBandCommsDisplay from '../../components/console/cronus/CronusSBandCommsDisplay'
import { shallow, mount, render} from 'enzyme';

const defaultValues = {
  sBand: {
    elevationGimbal: 60,
    azimuthGimbal: 2
  }
}

describe("Cronus S-Band communications display", () => {
  it('renders correctly', () => {
    const wrapper = mount(<CronusSBandCommsDisplay values={defaultValues} />)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})