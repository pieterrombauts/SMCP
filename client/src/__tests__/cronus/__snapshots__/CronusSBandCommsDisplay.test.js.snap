// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cronus S-Band communications display renders correctly 1`] = `
.c0 svg {
  width: 100%;
  height: 100%;
}

<Styled(UCronusSBandCommsDisplay)
  values={
    Object {
      "sBand": Object {
        "azimuthGimbal": 2,
        "elevationGimbal": 60,
      },
    }
  }
>
  <UCronusSBandCommsDisplay
    className="c0"
    values={
      Object {
        "sBand": Object {
          "azimuthGimbal": 2,
          "elevationGimbal": 60,
        },
      }
    }
  >
    <div
      className="c0"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 1300 990"
      >
        <image
          height={990}
          href="/media/Cronus_3.png"
          width={1300}
          x={0}
          y={0}
        />
        <rect
          height="385"
          rx="5"
          ry="5"
          style={
            Object {
              "fill": "#36485b",
              "stroke": "#2a3a3f",
              "strokeWidth": "5",
            }
          }
          width="450"
          x="450"
          y="380"
        />
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="45"
          x="38%"
          y="45%"
        >
          S-Band Status
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="38%"
          y="51%"
        >
          Active S-Band
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="51%"
        >
          1
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="38%"
          y="56%"
        >
          S-Band1 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="56%"
        >
          On
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="41%"
          y="59%"
        >
          Elevation Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="59%"
        >
          60
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="59%"
        >
          째
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="41%"
          y="62%"
        >
          Azimuth Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="62%"
        >
          2
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="62%"
        >
          째
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="38%"
          y="67%"
        >
          S-Band2 Power
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="67%"
        >
          Off
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="41%"
          y="70%"
        >
          Elevation Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="70%"
        >
          0.00
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="70%"
        >
          째
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="41%"
          y="73%"
        >
          Azimuth Gimbal
        </text>
        <text
          fill="#d6ab62"
          fontFamily="Verdana"
          fontSize="20"
          x="58%"
          y="73%"
        >
          0.00
        </text>
        <text
          fill="white"
          fontFamily="Verdana"
          fontSize="20"
          x="63%"
          y="73%"
        >
          째
        </text>
      </svg>
    </div>
  </UCronusSBandCommsDisplay>
</Styled(UCronusSBandCommsDisplay)>
`;
